### PR TITLE
Module2/task2

### PR DESCRIPTION
**Pull request by praveenthinkitive**
Task 2

### CODE REVIEW

File: Module2/day2/index2.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+});
+
+// Define the book schema
+const bookSchema = new mongoose.Schema({
+  name: String,
+  age: Number
+});
+
+// Register the model with Mongoose
+const bookModel = mongoose.model("books", bookSchema);
```
Comment (imtiyazthink) [inline Module2/day2/index2.js:79]: Modularise the code with proper folder structure, aviod writting all in one

File: Module2/task2/node_modules/.bin/mime
Comment (imtiyazthink) [inline Module2/task2/node_modules/.bin/mime:1]: Node Modules should be ignored, should not be pushed

File: Module2/task2/server.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+                fs.readFile(path.join(__dirname,'user.json'),'utf8',(err, data)=>{
+                    if(data){
+                        res.send(data);
+                    }
+                });
+            })
+        });
+    });
+
+    app.put('/users/:id', (req, res) => {
```
Comment (imtiyazthink) [inline Module2/task2/server.js:84]: No proper excetion handling is done in any of the file

File: Module2/task2/server.js [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        }     
+    });
+});
+
+app.delete('/users/:id', (req, res) => {
+    const userId = req.params.id;
+    // console.log(userId);
+    fs.readFile(path.join(__dirname, 'user.json'), 'utf8', (err, data) => {
+    const input=JSON.parse(data);
+    for (let i in input.users){
```
Comment (imtiyazthink) [inline Module2/task2/server.js:41]: Why we are not using Higher order function here?

**Review by imtiyazthink — CHANGES_REQUESTED** [review]
Code is so messy, didn't understood what to review.
No code modularisation, pushing node modules and env files
No proper crud operation is present
No exception handling 
No proper status code is followed

File: Module2/day2/.env [hunk 4]
```diff
@@ -0,0 +1,2 @@
+MDURL= 'mongodb://localhost:27017/bookstore'
```
[the remark this answers is not in the document]
Comment (praveenthinkitive) [reply Module2/day2/.env]: Removed env file

File: Module2/task2/node_modules/.bin/mime
Comment (praveenthinkitive) [reply Module2/task2/node_modules/.bin/mime:1]: Removed and ignored

File: Module2/task2/server.js (hunk 2 above)
Comment (praveenthinkitive) [reply Module2/task2/server.js]: Added exception handling in controller

File: Module2/task2/server.js (hunk 3 above)
Comment (praveenthinkitive) [reply Module2/task2/server.js]: Updated

**Pull request merged**